examples/agent_store/eval/bulk_generate.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+import asyncio
+from datetime import datetime
+from pathlib import Path
+
+import fire
+
+import pandas as pd
+from termcolor import cprint
+from tqdm import tqdm
+
+from ..api import AgentChoice, AgentStore
+
+MODEL = "meta-llama/Llama-3.1-8B-Instruct"
+
+
+async def app_bulk_generate(
+    host: str = "localhost",
+    port: int = 5000,
+    model: str = MODEL,
+    docs_dir: str = "",
+    dataset_path: str = "",
+):
+    df = pd.read_csv(dataset_path)
+    user_prompts = df["input_query"].tolist()
+
+    agent_store_app = AgentStore(host, port, model)
```

Comment:
are we expecting the entire app to be available as a single class? it seems like it will be better to just allow the app to whatever it wants but then prescribe the interface for how the generation is done. Goal would be allow us to do dependency injection at test time - stream in a set of app inputs (as a dataset which has all the columns needed that are specific to the app) and execute the same app generation code.

By taking a cue from regular test harnesses, we should have test-init which will setup the appropriate "state" before running the test. Inserting into memory could sit there.

Reply:
My expectation is:
1. app can do whatever it wants
2. we perform bulk generation using app code (similar to how we write unit tests, by invoking each function of app code)
3. this gives an example on how to run bulk_generation using existing AgentStore app on Memory agent, it will be different for Search agent

> dependency injection at test time - stream in a set of app inputs 

The bulk_generate script is doing: (1) taking in a dataset of user input_prompts (2) calling `agent_store_app.chat`, which is the code that gets called when user press enter within the app.